pallets/creditcoin/src/lib.rs

```diff
@@ -772,6 +782,13 @@ pub mod pallet {
 			let address = Self::get_address(&address_id)?;
 			ensure!(address.owner == who, Error::<T>::NotAddressOwner);
 
+			let currency =
+				Currencies::<T>::get(&terms.currency).ok_or(Error::<T>::CurrencyNotRegistered)?;
```

Comment:
I think there's no explicit unit test which asserts on `CurrencyNotRegistered` if you try `add_ask_order` with a currency which hasn't been registered first.